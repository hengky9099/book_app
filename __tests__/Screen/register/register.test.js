import * as React from 'react';
import Register from '../../../src/screens/register/index';
import '@testing-library/jest-dom/extend-expect';
import {create} from 'react-test-renderer';
import ContainerTesting from '../../../src/helpers/reduxTesting';

jest.mock('redux-persist', () => {
  const real = jest.requireActual('redux-persist');
  return {
    ...real,
    persistReducer: jest
      .fn()
      .mockImplementation((config, reducers) => reducers),
  };
});

describe('Register Testing', () => {
  describe('State Testing', () => {
    const component = create(ContainerTesting(<Register />));
    const root = component.root;
    const {register} = root.props.store.getState();
    test('state testing', () => {
      expect(root.props.store).toBeTruthy();
      expect(register).toBeTruthy();
      expect(typeof register.email).toEqual('string');
      expect(typeof register.password).toEqual('string');
      expect(typeof register.name).toEqual('string');
    });
  });

  describe('Snapshot Register', () => {
    test('Snapshot', () => {
      const tree = create(ContainerTesting(<Register />));
      expect(tree).toMatchSnapshot();
    });
  });
});
