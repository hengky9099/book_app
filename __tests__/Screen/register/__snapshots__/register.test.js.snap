// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register Testing Snapshot Register Snapshot 1`] = `
<RCTScrollView
  style={
    Object {
      "backgroundColor": "white",
    }
  }
>
  <View>
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/images/logo.png",
        }
      }
      style={
        Object {
          "height": 314.2857142857143,
          "marginLeft": 110,
          "marginTop": 157.14285714285714,
          "width": 314.2857142857143,
        }
      }
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Full Name"
      style={
        Object {
          "borderColor": "gray",
          "borderRadius": 15.714285714285714,
          "borderStyle": "solid",
          "borderWidth": 1.5714285714285714,
          "marginBottom": 31.428571428571427,
          "marginLeft": 50.285714285714285,
          "width": 440,
        }
      }
      testID="registration.username"
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Email"
      style={
        Object {
          "borderColor": "gray",
          "borderRadius": 15.714285714285714,
          "borderStyle": "solid",
          "borderWidth": 1.5714285714285714,
          "marginBottom": 31.428571428571427,
          "marginLeft": 50.285714285714285,
          "width": 440,
        }
      }
      textContentType="emailAddress"
      value=""
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Password"
      secureTextEntry={true}
      style={
        Object {
          "borderColor": "gray",
          "borderRadius": 15.714285714285714,
          "borderStyle": "solid",
          "borderWidth": 1.5714285714285714,
          "marginBottom": 31.428571428571427,
          "marginLeft": 50.285714285714285,
          "width": 440,
        }
      }
      value=""
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4D96FF",
          "borderRadius": 15.714285714285714,
          "height": 62.857142857142854,
          "justifyContent": "center",
          "marginLeft": 62.857142857142854,
          "opacity": 1,
          "width": 408.57142857142856,
        }
      }
      testID="RegisterButton"
    >
      <Text
        style={
          Object {
            "color": "white",
          }
        }
      >
        Register
      </Text>
    </View>
    <Text
      style={
        Object {
          "color": "gray",
          "fontSize": 22,
          "fontWeight": "bold",
          "marginLeft": 157.14285714285714,
          "marginTop": 15.714285714285714,
        }
      }
    >
      Already have an account?
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
      testID="LoginButton"
    >
      <Text
        style={
          Object {
            "color": "black",
            "fontSize": 25.142857142857142,
            "fontWeight": "bold",
            "left": 226.28571428571428,
            "marginTop": 15.714285714285714,
          }
        }
      >
        Login
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
